src/ui/pages/search/Search.jsx

```diff
@@ -1,239 +1,196 @@
-                    });
-
-                    return;
-                }
-            })
-        }
-
-        if (this.state.keyword) {
-            // try to get single result
-            WS.getSearchSingleResult(this.state.keyword)
-                .then(response => {
-                    if (response.body && response.body.data) {
-                        this.setState({
-                            redirectRoute: response.body.data.link
-                        });
-                    }
-                }).catch(console.error);
-        }
-    }
+    // redirects to the record selected with arrows
+    if (selectedItemIndex >= 0 && selectedItemIndex < results.length) {
+      setRedirectRoute(results[selectedItemIndex].link);
 
-    handleSearchArrowDown() {
-        if (this.state.selectedItemIndex !== this.state.results.length - 1) {
-            this.setState({
-                selectedItemIndex: ++this.state.selectedItemIndex
-            });
-        } else {
-            this.setState({
-                selectedItemIndex: 0
-            });
-        }
+      return;
     }
 
-    handleSearchArrowUp() {
-        if (this.state.selectedItemIndex > 0) {
-            this.setState(() => ({
-                selectedItemIndex: --this.state.selectedItemIndex
-            }));
-        } else {
-            this.setState(() => ({
-                selectedItemIndex: this.state.results.length - 1
-            }));
+    // if enter pressed and there is a record
+    // with the code exactly matching the keyword
+    // redirect to this record
+    if (results.length > 0) {
+      results.forEach((result) => {
+        if (result.code === keyword) {
+          setRedirectRoute(result.link);
+          return;
         }
+      });
     }
 
-    scrollWindowIfNecessary() {
-        let selectedRow = document.getElementsByClassName("selectedRow")[0];
-
-        if (!selectedRow) {
-            return;
-        }
-
-        let rect = selectedRow.getBoundingClientRect();
-        const margin = 230;
-        const isInViewport = rect.top >= margin &&
-            rect.bottom <= ((window.innerHeight || document.documentElement.clientHeight) - margin);
-
-        if (!isInViewport) {
-            selectedRow.scrollIntoView();
-        }
+    if (keyword) {
+      // try to get single result
+      WS.getSearchSingleResult(keyword)
+        .then((response) => {
+          if (response.body && response.body.data) {
+            setRedirectRoute(response.body.data.link);
+          }
+        })
+        .catch(handleError);
     }
+  };
 
-    fetchNewData(keyword, entityTypes) {
-        if (!!this.cancelSource) {
-            this.cancelSource.cancel();
-        }
-
-        if (!keyword) {
-            this.setState({
-                results: [],
-                keyword,
-                isFetching: false
-            })
-            return;
-        }
-
-        this.cancelSource = Ajax.getAxiosInstance().CancelToken.source();
-
```

Comment:
I just wanted to point out that I remove the logic with cancelSource - it wasn't working for me when I was using useRef to store that cancelSource value. I believe it was implemented to prevent race conditions, however now tanstack/react-query is handling that for us, so I believe it should not be a problem to remove this logic

Reply:
Okay, as I was talking with @DevauxJ , we should keep this behaviour. I will work on implementing this cancelling logic to the new version of components